### PR TITLE
Add Job Poster Builder Tasks page (component only)

### PR DESCRIPTION
**Pull request by Khristophor**
Affects #959

### CODE REVIEW

File: resources/assets/js/components/JobTasks/JobTasks.tsx [hunk 1]
```diff
@@ -1,222 +1,387 @@
...
+                                          description="Error message displayed when too many tasks are on screen."
+                                          defaultMessage="You have exceeded the maximum number of key tasks allowed, but that’s okay. You can continue to add key tasks as you brainstorm here, but you will be asked to trim your list to 6 key tasks or fewer to proceed."
+                                        />
+                                      </p>
+                                    </div>
+                                  </div>
+                                </div>
+                              )}
+                              <div
+                                key={`listItem${index}`}
```
Comment (tristan-orourke) [inline resources/assets/js/components/JobTasks/JobTasks.tsx]: Its dangerous to use index as the key, _particularly_ when the order will change. Is it possible to give new tasks a temporary unique id somehow? 
https://reactjs.org/docs/lists-and-keys.html#keys

File: resources/assets/js/components/JobTasks/JobTasks.tsx [hunk 2]
```diff
@@ -1,222 +1,387 @@
...
+                name="tasks"
+                render={(arrayHelpers): React.ReactElement => (
+                  <>
+                    <div data-c-grid="gutter">
+                      {values.tasks &&
+                        values.tasks.length > 0 &&
+                        values.tasks.map(
+                          (task, index): React.ReactElement => (
+                            <>
+                              {validCount! === index && (
```
Comment (tristan-orourke) [inline resources/assets/js/components/JobTasks/JobTasks.tsx]: Why the non-null assertions on validCount, here and later? Don't we need to account for it possibly being undefined?

File: resources/assets/js/components/JobTasks/JobTasks.tsx [hunk 3]
```diff
@@ -1,222 +1,387 @@
...
+                                  >
+                                    <i
+                                      className="fas fa-trash"
+                                      data-c-colour="stop"
+                                    />
+                                  </button>
+                                </div>
+                                <Field
+                                  id={`task${task.id}`}
+                                  name={`tasks.${index}.description`}
```
Comment (tristan-orourke) [inline resources/assets/js/components/JobTasks/JobTasks.tsx]: I would have expected using index like this not to work, but it tests fine in storybook. I guess arrayHelpers makes it work?

File: resources/assets/js/components/JobTasks/JobTasks.tsx (hunk 2 above)
Comment (Khristophor) [reply resources/assets/js/components/JobTasks/JobTasks.tsx]: After reviewing the functionality for the page, I'm making it required. I'll take out the non-nulls.

File: resources/assets/js/components/JobTasks/JobTasks.tsx (hunk 3 above)
Comment (Khristophor) [reply resources/assets/js/components/JobTasks/JobTasks.tsx]: It's a combo of Formik allowing it and having a helper function to grab the nested object (if it exists). I had to update the TextAreaInput component to make use of it, but it seems to work fine for array nested inputs as well as individual ones.

https://jaredpalmer.com/formik/docs/api/fieldarray#fieldarray-array-of-objects

File: resources/assets/js/components/JobTasks/JobTasks.tsx (hunk 1 above)
Comment (Khristophor) [reply resources/assets/js/components/JobTasks/JobTasks.tsx]: I'll see what I can do. We'll have to figure out some sort of temp ID standard, unless we're just worried about having it as the key in isolated cases like this.

Comment (Khristophor) [reply resources/assets/js/components/JobTasks/JobTasks.tsx]: As we discussed I added the [nanoid](https://github.com/ai/nanoid) library for generating temp IDs. We'll keep it to 10 characters for now as that seems to be more than fine.

I'm still seeing console warnings for missing keys inside of a loop, so I'm not sure if we'll still see unnecessary rerenders inside the tree. I think it has to do with how I'm using fragments inside the FieldArray but no combination of adding keys to any element inside of those loops seem to resolve it.

**Pull request merged**